### PR TITLE
Do not crash when cannot verifying LE user

### LINKED ISSUE #185: Timed out from Let's Encrypt made kube-lego crash
I just noticed that my kube-lego container crash a lot. I have this log line just when it crash :
```
time="2017-05-19T09:45:40Z" level=fatal msg="fatal error verifying existing user: failed to retrieve ACME account for URI 'https://acme-v01.api.letsencrypt.org/acme/reg/8656258': 504 : 504 Gateway Time-out" context=acme 
```
It happens just after it try a certificate request for a domain which not (yet) point to my server. I don't know if :
- it fails because of a Let's Encrypt side issue (HTTP 504)
- it fails because my domain is not pointing to my server

In any case, I think it is a problem that the container crash if it run into an HTTP504.

### PR DESCRIPTION
**Pull request by pichouk**
Try to fix #185 
Return an error when cannot verifying LE user. I'm not go developer, maybe my patch is not working :smile:

### CODE REVIEW

**Comment by pichouk** [comment]
Up ? :)

File: pkg/acme/cert_request.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ func (a *Acme) ensureAcmeClient() error {
 	account, err := a.validateUser(client, accountURI)
 	if err != nil {
 		a.Log().Fatalf("fatal error verifying existing user: %s", err)
```
Comment (munnerz) [inline pkg/acme/cert_request.go]: The `Fatal` log functions will automatically call `os.Exit(1)` after their execution, so the new `return err` line is actually unreachable. Can you change the `Fatalf` line to call `Errorf` instead? This should hopefully fix your issue!

**Comment by pichouk** [comment]
I'll try to test but honestly I'm not at all a Go developer. I have a lot of problems understanding and make working the Go dependencies system, so I cannot build the image.

**Comment by pichouk** [comment]
I build the [`pichouk/lego-test`](https://hub.docker.com/r/pichouk/lego-test/) Docker image to test. Seems OK but cannot reproduce since, with an invalid ACME server, kube-lego get an error before reaching this point. I'll see if I can setup a fake server which return HTTP 504 (like in #185 )

**Comment by pichouk** [comment]
Closing, the repository seems inactive and I think it will never be merged.

**Pull request closed without merging**